pkg/fanotify/conn/conn.go

```diff
@@ -8,16 +8,32 @@ package conn
 
 import (
 	"bufio"
+	"encoding/json"
 	"io"
 )
 
 type Client struct {
-	Scanner *bufio.Scanner
+	Reader *bufio.Reader
 }
 
-func (c *Client) GetPath() (string, error) {
-	if !c.Scanner.Scan() { // NOTE: no timeout
-		return "", io.EOF
+type EventInfo struct {
+	Path      string `json:"path"`
+	Size      uint32 `json:"size"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+func (c *Client) GetEventInfo() ([]EventInfo, error) {
+	// Before reaching '\n', the reader has successfully read
+	// the event information from optimizer server.
+	data, err := c.Reader.ReadBytes('\n')
```

Comment:
Generally, we should not assume what is the special token in the stream. Let's merge this PR now and I will try to improve it later.

Reply:
Got it.